### PR TITLE
Added info to Password Rest in Users.md to convert a user to a federated user

### PR DESCRIPTION
**Pull request by eklein-okta**
@stsai-okta @karlmcguinness-okta @croche-okta 

This is the doc for OKTA-53810 going into preview on April 8. If this PR is good, we can have a link to it from the preview release notes. Thx!  -Evan

### CODE REVIEW

File: docs/api/rest/users.md [hunk 1]
```diff
@@ -1978,6 +1980,30 @@ curl -v -H "Authorization: SSWS yourtoken" \
 }
 ~~~
 
+##### Request Example (Convert a User to a Federated User)
+{:.api .api-request .api-request-example}
+
+To convert a user to a federated user, pass `FEDERATED` as the ***type*** in the [Provider Object](#provider-object). The `sendEmail`
+parameter must be false or omitted for this type of conversion.
+
+~~~ ruby
```
Comment (stsai-okta) [inline docs/api/rest/users.md]: are we using ruby to display all of the curls?

Comment (eklein-okta) [inline docs/api/rest/users.md]: That's how that whole doc is done...

File: docs/api/rest/users.md [hunk 2]
```diff
@@ -1978,6 +1980,30 @@ curl -v -H "Authorization: SSWS yourtoken" \
...
+curl -v -H "Authorization: SSWS yourtoken" \
+-H "Accept: application/json" \
+-H "Content-Type: application/json" \
+-X POST "https://your-domain.okta.com/api/v1/users/00ub0oNGTSWTBKOLGLNR/?provider=FEDERATION&sendEmail=false"
```
Comment (stsai-okta) [inline docs/api/rest/users.md]: This is the wrong url: 
https://your-domain.okta.com/api/v1/user/00ub0oNGTSWTBKOLGLNR/lifecycle/reset_password?provider=FEDERATION&sendEmail=false

File: docs/api/rest/users.md [hunk 3]
```diff
@@ -1940,6 +1940,8 @@ Generates a one-time token (OTT) that can be used to reset a user's password.  T
 
 This operation will transition the user to the status of `RECOVERY` and the user will not be able to login or initiate a forgot password flow until they complete the reset flow.
 
+**Note:** You can also use this API to convert a user with the Okta Credential Provider to a use with a Federated Provider. After this conversion, the user cannot directly sign in with password. The second example demonstrate this usage.
```
Comment (stsai-okta) [inline docs/api/rest/users.md]: should probably be ```to use a Federated Provider``` not ```to a use with a Federated Provider```

File: docs/api/rest/users.md [hunk 4]
```diff
@@ -1978,6 +1980,30 @@ curl -v -H "Authorization: SSWS yourtoken" \
 }
 ~~~
 
+##### Request Example (Convert a User to a Federated User)
+{:.api .api-request .api-request-example}
+
+To convert a user to a federated user, pass `FEDERATED` as the ***type*** in the [Provider Object](#provider-object). The `sendEmail`
```
Comment (stsai-okta) [inline docs/api/rest/users.md]: should we use ```provider``` instead of ```type```?

**Comment by eklein-okta** [comment]
@stsai-okta All your comments are in the doc.

**Comment by stsai-okta** [comment]
looks good to me.  Please squash your commits then :goberserk:

**Pull request merged**